### ISSUE TITLE
npm test fails

### ISSUE THREAD

**Issue by dylanb** [angular/protractor#2514]
Using OS X and Selenium 2.47, I get the results below when running `npm test`. I have Firefox 40.0.3 installed as the default Firefox. Some of the tests cause the Firefox browser to seg fault (I believe this is the test that tries to install a Firefox profile).

One of the failing Chrome tests (testing for an alert box) does leave a browser hanging with an alert showing

```
Dylans-MacBook-Pro-2:protractor dylanbarrell$ npm test

> protractor@2.2.0 pretest /Users/dylanbarrell/private/protractor
> jshint lib spec scripts


> protractor@2.2.0 test /Users/dylanbarrell/private/protractor
> node scripts/test.js

running: node lib/cli.js spec/basicConf.js
Using the selenium server at http://localhost:4444/wd/hub
[launcher] Running 1 instances of WebDriver
Started
.....WARNING - more than one element found for locator by.binding("item.reusedBinding") - the first result will be used
..............F......................................................................................................F
/Users/dylanbarrell/private/protractor/node_modules/selenium-webdriver/lib/atoms/error.js:113
  var template = new Error(this.message);
                 ^
NoSuchAlertError: no alert open
  (Session info: chrome=45.0.2454.99)
  (Driver info: chromedriver=2.12.301326 (093c7e07b4a916b690e784b0374c7f618f1ea4be),platform=Mac OS X 10.10.5 x86_64) (WARNING: The server did not provide any stacktrace information)
Command duration or timeout: 1 milliseconds
Build info: version: '2.47.1', revision: '411b314', time: '2015-07-30 03:03:16'
System info: host: 'Dylans-MacBook-Pro-2.local', ip: '192.168.1.3', os.name: 'Mac OS X', os.arch: 'x86_64', os.version: '10.10.5', java.version: '1.8.0_25'
Driver info: org.openqa.selenium.chrome.ChromeDriver
Capabilities [{applicationCacheEnabled=false, rotatable=false, mobileEmulationEnabled=false, chrome={userDataDir=/var/folders/0b/2xvqghmx4mq690kwq182h2h40000gn/T/.org.chromium.Chromium.MnLXzj}, takesHeapSnapshot=true, databaseEnabled=false, handlesAlerts=true, version=45.0.2454.99, platform=MAC, browserConnectionEnabled=false, nativeEvents=true, acceptSslCerts=true, locationContextEnabled=true, webStorageEnabled=true, browserName=chrome, takesScreenshot=true, javascriptEnabled=true, cssSelectorsEnabled=true}]
Session ID: c395832115824285c146bff90b4ef502
    at new bot.Error (/Users/dylanbarrell/private/protractor/node_modules/selenium-webdriver/lib/atoms/error.js:113:18)
    at Object.bot.response.checkResponse (/Users/dylanbarrell/private/protractor/node_modules/selenium-webdriver/lib/atoms/response.js:106:9)
    at /Users/dylanbarrell/private/protractor/node_modules/selenium-webdriver/lib/webdriver/webdriver.js:377:20
    at [object Object].promise.ControlFlow.runInFrame_ (/Users/dylanbarrell/private/protractor/node_modules/selenium-webdriver/lib/webdriver/promise.js:1877:20)
    at [object Object].promise.Callback_.goog.defineClass.notify (/Users/dylanbarrell/private/protractor/node_modules/selenium-webdriver/lib/webdriver/promise.js:2464:25)
    at [object Object].promise.Promise.notify_ (/Users/dylanbarrell/private/protractor/node_modules/selenium-webdriver/lib/webdriver/promise.js:563:12)
    at Array.forEach (native)
    at Object.goog.array.forEach (/Users/dylanbarrell/private/protractor/node_modules/selenium-webdriver/lib/goog/array/array.js:203:43)
    at [object Object].promise.Promise.notifyAll_ (/Users/dylanbarrell/private/protractor/node_modules/selenium-webdriver/lib/webdriver/promise.js:552:16)
    at goog.async.run.processWorkQueue (/Users/dylanbarrell/private/protractor/node_modules/selenium-webdriver/lib/goog/async/run.js:125:21)
From: Task: Run it("should deal with alerts") in control flow
    at Object.<anonymous> (/Users/dylanbarrell/private/protractor/node_modules/jasminewd2/index.js:81:14)
    at attemptAsync (/Users/dylanbarrell/private/protractor/node_modules/jasmine/node_modules/jasmine-core/lib/jasmine-core/jasmine.js:1819:24)
    at QueueRunner.run (/Users/dylanbarrell/private/protractor/node_modules/jasmine/node_modules/jasmine-core/lib/jasmine-core/jasmine.js:1774:9)
    at /Users/dylanbarrell/private/protractor/node_modules/jasmine/node_modules/jasmine-core/lib/jasmine-core/jasmine.js:1801:16
    at /Users/dylanbarrell/private/protractor/node_modules/jasmine/node_modules/jasmine-core/lib/jasmine-core/jasmine.js:1745:9
    at /Users/dylanbarrell/private/protractor/node_modules/jasminewd2/index.js:18:5
    at promise.ControlFlow.runInFrame_ (/Users/dylanbarrell/private/protractor/node_modules/selenium-webdriver/lib/webdriver/promise.js:1877:20)
    at promise.Callback_.goog.defineClass.notify (/Users/dylanbarrell/private/protractor/node_modules/selenium-webdriver/lib/webdriver/promise.js:2464:25)
    at promise.Promise.notify_ (/Users/dylanbarrell/private/protractor/node_modules/selenium-webdriver/lib/webdriver/promise.js:563:12)
    at Array.forEach (native)
From asynchronous test:
Error
    at Suite.<anonymous> (/Users/dylanbarrell/private/protractor/spec/basic/navigation_spec.js:8:3)
    at addSpecsToSuite (/Users/dylanbarrell/private/protractor/node_modules/jasmine/node_modules/jasmine-core/lib/jasmine-core/jasmine.js:743:25)
    at Env.describe (/Users/dylanbarrell/private/protractor/node_modules/jasmine/node_modules/jasmine-core/lib/jasmine-core/jasmine.js:713:7)
    at jasmineInterface.describe (/Users/dylanbarrell/private/protractor/node_modules/jasmine/node_modules/jasmine-core/lib/jasmine-core/jasmine.js:3219:18)
    at Object.<anonymous> (/Users/dylanbarrell/private/protractor/spec/basic/navigation_spec.js:3:1)
    at Module._compile (module.js:430:26)
    at Object.Module._extensions..js (module.js:448:10)
    at Module.load (module.js:355:32)
    at Function.Module._load (module.js:310:12)
[launcher] Process exited with error code 1
>>> fail: Error: expecting exit code: 0, actual: 1
running: node lib/cli.js spec/multiConf.js
>>> pass
running: node lib/cli.js spec/altRootConf.js
>>> pass
running: node lib/cli.js spec/onCleanUpAsyncReturnValueConf.js
>>> pass
running: node lib/cli.js spec/onCleanUpNoReturnValueConf.js
>>> pass
running: node lib/cli.js spec/onCleanUpSyncReturnValueConf.js
>>> pass
running: node lib/cli.js spec/onPrepareConf.js
>>> pass
running: node lib/cli.js spec/onPrepareFileConf.js
>>> pass
running: node lib/cli.js spec/onPreparePromiseConf.js
>>> pass
running: node lib/cli.js spec/onPreparePromiseFileConf.js
>>> pass
running: node lib/cli.js spec/mochaConf.js
>>> pass
running: node lib/cli.js spec/cucumberConf.js
>>> pass
running: node lib/cli.js spec/withLoginConf.js
>>> pass
running: node lib/cli.js spec/suitesConf.js --suite okmany
>>> pass
running: node lib/cli.js spec/suitesConf.js --suite okspec
>>> pass
running: node lib/cli.js spec/suitesConf.js --suite okmany,okspec
>>> pass
running: node lib/cli.js spec/plugins/smokeConf.js
>>> pass
running: node lib/cli.js spec/plugins/multiPluginConf.js
>>> pass
running: node lib/cli.js spec/plugins/jasmine1PostTestConf.js
>>> pass
running: node lib/cli.js spec/plugins/jasmine2PostTestConf.js
>>> pass
running: node lib/cli.js spec/plugins/mochaPostTestConf.js
>>> pass
running: node lib/cli.js spec/plugins/cucumberPostTestConf.js
>>> pass
running: node lib/cli.js spec/plugins/browserGetSyncedConf.js
>>> pass
running: node lib/cli.js spec/plugins/browserGetUnsyncedConf.js
>>> pass
running: node lib/cli.js spec/plugins/waitForAngularConf.js
>>> pass
running: node lib/cli.js spec/interactionConf.js
>>> pass
running: node lib/cli.js spec/directConnectConf.js
[launcher] Running 2 instances of WebDriver

------------------------------------
[chrome #1] PID: 64295
[chrome #1] Specs: /Users/dylanbarrell/private/protractor/spec/directConnect/directconnect_spec.js
[chrome #1]

[launcher] Runner process exited unexpectedly with error code: 1
[launcher] 1 instance(s) of WebDriver still running

------------------------------------
[firefox #2] PID: 64296
[firefox #2] Specs: /Users/dylanbarrell/private/protractor/spec/directConnect/directconnect_spec.js
[firefox #2]
[firefox #2] Using FirefoxDriver directly...
[firefox #2]
[firefox #2] /Users/dylanbarrell/private/protractor/node_modules/selenium-webdriver/lib/webdriver/promise.js:1877
[firefox #2]       var result = fn();
[firefox #2]                    ^
[firefox #2] Error: Failed to install profile; firefox terminated with Result(code=null, signal=SIGSEGV)
[firefox #2]     at Error (native)
[firefox #2]     at /Users/dylanbarrell/private/protractor/node_modules/selenium-webdriver/firefox/binary.js:239:67
[firefox #2]     at [object Object].promise.ControlFlow.runInFrame_ (/Users/dylanbarrell/private/protractor/node_modules/selenium-webdriver/lib/webdriver/promise.js:1877:20)
[firefox #2]     at [object Object].promise.Callback_.goog.defineClass.notify (/Users/dylanbarrell/private/protractor/node_modules/selenium-webdriver/lib/webdriver/promise.js:2464:25)
[firefox #2]     at [object Object].promise.Promise.notify_ (/Users/dylanbarrell/private/protractor/node_modules/selenium-webdriver/lib/webdriver/promise.js:563:12)
[firefox #2]     at Array.forEach (native)
[firefox #2]     at Object.goog.array.forEach (/Users/dylanbarrell/private/protractor/node_modules/selenium-webdriver/lib/goog/array/array.js:203:43)
[firefox #2]     at [object Object].promise.Promise.notifyAll_ (/Users/dylanbarrell/private/protractor/node_modules/selenium-webdriver/lib/webdriver/promise.js:552:16)
[firefox #2]     at goog.async.run.processWorkQueue (/Users/dylanbarrell/private/protractor/node_modules/selenium-webdriver/lib/goog/async/run.js:125:21)
[firefox #2]     at process._tickCallback (node.js:357:9)
[firefox #2] From: Task: WebDriver.createSession()
[firefox #2]     at Function.webdriver.WebDriver.acquireSession_ (/Users/dylanbarrell/private/protractor/node_modules/selenium-webdriver/lib/webdriver/webdriver.js:155:22)
[firefox #2]     at Function.webdriver.WebDriver.createSession (/Users/dylanbarrell/private/protractor/node_modules/selenium-webdriver/lib/webdriver/webdriver.js:129:30)
[firefox #2]     at new Driver (/Users/dylanbarrell/private/protractor/node_modules/selenium-webdriver/firefox/index.js:272:36)
[firefox #2]     at [object Object].DirectDriverProvider.getNewDriver (/Users/dylanbarrell/private/protractor/lib/driverProviders/direct.js:75:16)
[firefox #2]     at [object Object].Runner.createBrowser (/Users/dylanbarrell/private/protractor/lib/runner.js:180:37)
[firefox #2]     at /Users/dylanbarrell/private/protractor/lib/runner.js:256:21
[firefox #2]     at _fulfilled (/Users/dylanbarrell/private/protractor/node_modules/q/q.js:797:54)
[firefox #2]     at self.promiseDispatch.done (/Users/dylanbarrell/private/protractor/node_modules/q/q.js:826:30)
[firefox #2]     at Promise.promise.promiseDispatch (/Users/dylanbarrell/private/protractor/node_modules/q/q.js:759:13)
[firefox #2]     at /Users/dylanbarrell/private/protractor/node_modules/q/q.js:525:49

[launcher] Runner process exited unexpectedly with error code: 1
[launcher] 0 instance(s) of WebDriver still running
[launcher] chrome #1 failed with exit code: 1
[launcher] firefox #2 failed with exit code: 1
[launcher] overall: 2 process(es) failed to complete
[launcher] Process exited with error code 100
>>> fail: Error: expecting exit code: 0, actual: 100
running: node lib/cli.js spec/restartBrowserBetweenTestsConf.js
>>> pass
running: node lib/cli.js spec/getCapabilitiesConf.js
>>> pass
running: node lib/cli.js spec/controlLockConf.js
>>> pass
running: node lib/cli.js spec/customFramework.js
>>> pass
running: node node_modules/.bin/jasmine JASMINE_CONFIG_PATH=scripts/unit_test.json
module.js:338
    throw err;
          ^
Error: Cannot find module 'marked'
    at Function.Module._resolveFilename (module.js:336:15)
    at Function.Module._load (module.js:286:25)
    at Module.require (module.js:365:17)
    at require (module.js:384:17)
    at Object.<anonymous> (/Users/dylanbarrell/private/protractor/website/docgen/processors/tag-fixer.js:2:14)
    at Module._compile (module.js:430:26)
    at Object.Module._extensions..js (module.js:448:10)
    at Module.load (module.js:355:32)
    at Function.Module._load (module.js:310:12)
    at Module.require (module.js:365:17)
>>> fail: Error: expecting exit code: 0, actual: 1
running: node scripts/interactive_tests/interactive_test.js

/Users/dylanbarrell/private/protractor/node_modules/q/q.js:126
                    throw e;
                          ^
Did not start interactive server in 10s. Server startup output: /Users/dylanbarrell/private/protractor/node_modules/q/q.js:126
                    throw e;
                          ^
Error: No selenium server jar found at the specified location (/Users/dylanbarrell/private/protractor/selenium/selenium-server-standalone-2.45.0.jar). Check that the version number is up to date.
    at LocalDriverProvider.addDefaultBinaryLocs_ (/Users/dylanbarrell/private/protractor/lib/driverProviders/local.js:37:11)
    at LocalDriverProvider.setupEnv (/Users/dylanbarrell/private/protractor/lib/driverProviders/local.js:71:8)
    at Runner.run (/Users/dylanbarrell/private/protractor/lib/runner.js:254:31)
    at /Users/dylanbarrell/private/protractor/lib/launcher.js:171:21
    at _fulfilled (/Users/dylanbarrell/private/protractor/node_modules/q/q.js:797:54)
    at self.promiseDispatch.done (/Users/dylanbarrell/private/protractor/node_modules/q/q.js:826:30)
    at Promise.promise.promiseDispatch (/Users/dylanbarrell/private/protractor/node_modules/q/q.js:759:13)
    at /Users/dylanbarrell/private/protractor/node_modules/q/q.js:573:44
    at flush (/Users/dylanbarrell/private/protractor/node_modules/q/q.js:108:17)
    at doNTCallback0 (node.js:408:9)

>>> fail: Error: expecting exit code: 0, actual: 1
running: node scripts/interactive_tests/with_base_url.js

/Users/dylanbarrell/private/protractor/node_modules/q/q.js:126
                    throw e;
                          ^
Did not start interactive server in 10s. Server startup output: /Users/dylanbarrell/private/protractor/node_modules/q/q.js:126
                    throw e;
                          ^
Error: No selenium server jar found at the specified location (/Users/dylanbarrell/private/protractor/selenium/selenium-server-standalone-2.45.0.jar). Check that the version number is up to date.
    at LocalDriverProvider.addDefaultBinaryLocs_ (/Users/dylanbarrell/private/protractor/lib/driverProviders/local.js:37:11)
    at LocalDriverProvider.setupEnv (/Users/dylanbarrell/private/protractor/lib/driverProviders/local.js:71:8)
    at Runner.run (/Users/dylanbarrell/private/protractor/lib/runner.js:254:31)
    at /Users/dylanbarrell/private/protractor/lib/launcher.js:171:21
    at _fulfilled (/Users/dylanbarrell/private/protractor/node_modules/q/q.js:797:54)
    at self.promiseDispatch.done (/Users/dylanbarrell/private/protractor/node_modules/q/q.js:826:30)
    at Promise.promise.promiseDispatch (/Users/dylanbarrell/private/protractor/node_modules/q/q.js:759:13)
    at /Users/dylanbarrell/private/protractor/node_modules/q/q.js:573:44
    at flush (/Users/dylanbarrell/private/protractor/node_modules/q/q.js:108:17)
    at doNTCallback0 (node.js:408:9)

>>> fail: Error: expecting exit code: 0, actual: 1
running: node node_modules/minijasminenode/bin/minijn plugins/timeline/spec/unit.js
>>> pass
running: node lib/cli.js plugins/timeline/spec/conf.js
>>> pass
running: node lib/cli.js plugins/ngHint/spec/successConfig.js
>>> pass
running: node lib/cli.js plugins/accessibility/spec/successConfig.js
>>> pass
running: node lib/cli.js plugins/console/spec/consolePassConfig.js
>>> pass
running: node lib/cli.js spec/errorTest/singleFailureConf.js
>>> pass
running: node lib/cli.js spec/errorTest/timeoutConf.js
>>> pass
running: node lib/cli.js spec/errorTest/afterLaunchChangesExitCodeConf.js
>>> pass
running: node lib/cli.js spec/errorTest/multiFailureConf.js
>>> pass
running: node lib/cli.js spec/errorTest/shardedFailureConf.js
>>> pass
running: node lib/cli.js spec/errorTest/mochaFailureConf.js
>>> pass
running: node lib/cli.js spec/errorTest/pluginsFailingConf.js
>>> pass
running: node lib/cli.js spec/errorTest/slowHttpAndTimeoutConf.js
Using the selenium server at http://localhost:4444/wd/hub
[launcher] Running 1 instances of WebDriver
Started
FA Jasmine spec timed out. Resetting the WebDriver Control Flow.
F

Failures:
1) slow asynchronous events waits for http calls
  Message:
    Failed: Timed out waiting for Protractor to synchronize with the page after 1 seconds. Please see https://github.com/angular/protractor/blob/master/docs/faq.md
  Stack:
    Error: Failed: Timed out waiting for Protractor to synchronize with the page after 1 seconds. Please see https://github.com/angular/protractor/blob/master/docs/faq.md

2) slow asynchronous events waits for $timeout
  Message:
    Failed: Timed out waiting for Protractor to synchronize with the page after 1 seconds. Please see https://github.com/angular/protractor/blob/master/docs/faq.md. The following tasks were pending:
    - $timeout: function () {
          $scope.slowAngularTimeoutStatus = 'done';
        }
  Stack:
    Error: Failed: Timed out waiting for Protractor to synchronize with the page after 1 seconds. Please see https://github.com/angular/protractor/blob/master/docs/faq.md. The following tasks were pending:
    - $timeout: function () {
          $scope.slowAngularTimeoutStatus = 'done';
        }

2 specs, 2 failures
Finished in 4.469 seconds
[launcher] 0 instance(s) of WebDriver still running
[launcher] chrome #1 failed 2 test(s)
[launcher] overall: 2 failed spec(s)
[launcher] Process exited with error code 1
>>> fail: Error: did not fail with expected error with message: [The following tasks were pending[\s\S]*\$http: /slowcall]
running: node lib/cli.js plugins/ngHint/spec/failureConfig.js
>>> pass
running: node lib/cli.js plugins/accessibility/spec/failureConfig.js
>>> pass
running: node lib/cli.js plugins/console/spec/consoleFailConfig.js
>>> pass
running: node lib/cli.js plugins/console/spec/consoleFailErrorConfig.js
>>> pass
running: node lib/cli.js plugins/console/spec/consoleFailWarningConfig.js
>>> pass
running: node lib/cli.js plugins/console/spec/consoleFailFilterConfig.js
>>> pass
Summary: fail
npm ERR! Test failed.  See above for more details.
```

**Comment by surfjedi** [comment]
+1 I also have huge variation in results of tests when running from "protractor conf.js", or from the same command from npm test they half dont pass when run from npm.. but they pass when run other wise

**Comment by juliemr** [comment]
Confirmed, a couple of things have gotten stale here.

**Comment by juliemr** [comment]
The lingering firefox error will be solved by upgrading to `selenium-webdriver@2.47.1`

**Comment by juliemr** [comment]
Closing since the root cause is now limited to https://github.com/angular/protractor/issues/2245. Please follow along there. Thanks!
